packages/coinstac-api-server/src/auth-helpers.js

```diff
@@ -69,7 +69,29 @@ const helperFunctions = {
    */
   getUserDetails(credentials) {
     return helperFunctions.getRethinkConnection()
-    .then(connection => rethink.table('users').get(credentials.username).run(connection))
+    .then(connection => rethink.table('users').get(credentials.username).merge(user =>
+      ({
+        permissions: user('permissions').coerceTo('array')
```

Comment:
are all these coercing neccessary? Why isnt the raw output fine? Does this happen dbside or after?

Reply:
In the DB, a user currently has a *permissions* attribute as follows:

```json
"permissions": {
  "computations": {
    "multi-shot-test-id": [
      "owner"
    ] ,
    "single-shot-test-id": [
      "owner"
    ]
  } ,
  "consortia": {
    "test-cons-1": [
      "owner"
    ] ,
    "test-cons-2": [
      "member"
    ]
  } ,
  "pipelines": { }
}
```

The attribute is an object and we wanted to iterate through all "tables" - computations, consortia, pipelines - in the permissions object in such a way that we didn't have to hard code any table names in case we decided to add/remove others later. Making it an array allowed us to do that within the confines of the **rethink** api. This is really just to get at the "roles" in each table document and to swap them for their respective "verbs" as we discussed a while back. After the verbs are swapped-in, the resulting data is coerced back into an object.

So, to answer your question, it is saved as an object in the DB. But when it is queried, we alter the contents of that object by first transforming it into a series of arrays and swapping out the necessary values, and finally we build it back up into an object and push it out to the user. I wouldn't doubt there is an easier way to go about it, but I wasn't aware of it at the time.